vaadinfrontend/src/main/java/life/qbic/datamanager/views/notifications/NotificationDialog.java

```diff
@@ -0,0 +1,59 @@
+package life.qbic.datamanager.views.notifications;
+
+import com.vaadin.flow.component.HasOrderedComponents;
+import com.vaadin.flow.component.confirmdialog.ConfirmDialog;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.H2;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+
+/**
+ * A dialog notifying the user of some event.
+ * <p>
+ * By default, this dialog comes with an icon and a text in the header. You can modify the header
+ * text with {@link #setTitle}. When setting the icon with {@link #setHeaderIcon}, you can modify
+ * the color of the icon by assigning the following css classes:
+ * <ul>
+ *   <li>error-icon</li>
+ *   <li>warning-icon</li>
+ *   <li>info-icon</li>
+ * </ul>
+ * <p>
+ * The content of the dialog can be accessed by extending classes in the final field {@link #content}.
+ */
+public class NotificationDialog extends ConfirmDialog {
+
+  private final H2 title;
```

Comment:
NIT: 
Technically you're introducing vaadin dependent styling here. Maybe it's easier to specify the css values you want in the .title class

Reply:
No, I introduce a `h2` dom element here which is styled as specified in the css.

Reply:
If you check the vaadin-custom.css we already had to adapt the H2 element since vaadin changed the margins from 23 to 24, which is why it's always better to not be reliant on their components. 

/* Margins of H2 Elements have changed from Vaadin 23 to Vaadin 24*/
h2 {
  margin-bottom: 0.5em;
  margin-top: 1.25em;
}